ggu/main.c

```diff
@@ -175,11 +187,30 @@ process_single_packet(struct rte_mbuf *pkt, const struct ggu_config *ggu_conf)
 		if (memcmp(ip6hdr->dst_addr,
 				ggu_conf->net->back.ip6_addr.s6_addr,
 				sizeof(ip6hdr->dst_addr)) != 0) {
+			RTE_LOG(NOTICE, GATEKEEPER,
+				"ggu: received an IPv6 packet destinated to other host!\n");
 			return;
 		}
 
-		udphdr = (struct udp_hdr *)&ip6hdr[1];
+		nexthdr = ip6hdr->proto;
+		remaining_len -= sizeof(struct ipv6_hdr);
```

Comment:
Why split up the computation of `remaining_len` into three separate assignment statements in this function? You might not even need a separate variable at all for it.